e2e_tests/tests/per_provider/normal_tests/asym_sign_verify.rs

```diff
@@ -891,33 +899,22 @@ fn verify_message_not_permitted() {
     assert_eq!(error, ResponseStatus::PsaErrorNotPermitted);
 }
 
-#[cfg(feature = "tpm-provider")]
 #[test]
 fn wildcard_hash_not_supported() {
-    let key_name = auto_test_keyname!();
     let mut client = TestClient::new();
-    let mut usage_flags: UsageFlags = Default::default();
-    let _ = usage_flags.set_sign_hash().set_verify_hash();
-
-    assert_eq!(
-        client
-            .generate_key(
-                key_name,
-                Attributes {
-                    lifetime: Lifetime::Persistent,
-                    key_type: Type::RsaKeyPair,
-                    bits: 1024,
-                    policy: Policy {
-                        usage_flags,
-                        permitted_algorithms: Algorithm::AsymmetricSignature(
-                            AsymmetricSignature::RsaPkcs1v15Sign {
-                                hash_alg: SignHash::Any,
-                            },
-                        ),
-                    },
-                }
-            )
-            .unwrap_err(),
-        ResponseStatus::PsaErrorNotSupported
-    );
+
+    let mut attributes = TestClient::default_sign_rsa_attrs();
+    attributes.policy.permitted_algorithms =
+        Algorithm::AsymmetricSignature(AsymmetricSignature::RsaPkcs1v15Sign {
+            hash_alg: SignHash::Any,
+        });
+
+    if client.is_operation_supported(Opcode::CanDoCrypto) {
+        assert_eq!(
+            client
+                .can_do_crypto(CheckType::Generate, attributes)
+                .unwrap_err(),
+            ResponseStatus::PsaErrorNotSupported
+        );
+    }
```

Comment:
I wonder if this test should be moved to the file for key generation tests.

Reply:
It can be moved into key_attributes.rs, but as we test the signature algorithm, there is nothing wrong to keep it where it is I think.